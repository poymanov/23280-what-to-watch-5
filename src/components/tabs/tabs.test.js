import React from "react";
import renderer from "react-test-renderer";
import Tabs from "./tabs";
import {Router as BrowserRouter} from "react-router-dom";
import browserHistory from "../../etc/browser-history";

const movie = {
  id: 1,
  name: `test`,
  posterImage: `test`,
  previewImage: `test`,
  backgroundImage: `test`,
  backgroundColor: `test`,
  videoLink: `test`,
  previewVideoLink: `test`,
  description: `test`,
  rating: 1,
  scoresCount: 1,
  director: `test`,
  starring: [`test`],
  runTime: 1,
  genre: `test`,
  released: 1,
  isFavorite: false,
};

it(`Should Tabs render correctly`, () => {
  const tree = renderer
    .create(<BrowserRouter history={browserHistory}><Tabs movie={movie}/></BrowserRouter>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
