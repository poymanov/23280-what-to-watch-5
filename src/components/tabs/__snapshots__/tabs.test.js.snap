// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Tabs render correctly 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      1
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Bad
      </span>
      <span
        className="movie-rating__count"
      >
        1
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      test
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        test
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        test
      </strong>
    </p>
  </div>,
]
`;
